lib/wallaby/browser.ex

```diff
@@ -903,6 +903,65 @@ defmodule Wallaby.Browser do
   end
 
   @doc """
+  Checks if `query` is present within `parent` and raises if not found.
+
+  Returns the given `parent` if the assertion is correct so that it is easily
+  pipeable.
+
+  ## Examples
+
+      session
+      |> visit("/")
+      |> assert_has(Query.css(".login-button"))
+  """
+  @spec assert_has(parent, Query.t) :: parent
+
+  defmacro assert_has(parent, query) do
+    quote do
+      parent = unquote(parent)
+      query  = unquote(query)
+
+      case has?(parent, query) do
+        true ->
+          parent
+        false ->
+          raise Wallaby.ExpectationNotMet,
+                Query.ErrorMessage.message(query, :not_found)
+
+      end
+    end
+  end
+
+  @doc """
+  Checks if `query` is not present within `parent` and raises if it is found.
+
+  Returns the given `parent` if the query is not found so that it is easily
+  pipeable.
+
+  ## Examples
+
+      session
+      |> visit("/")
+      |> refute_has(Query.css(".secret-admin-content"))
+  """
+  @spec refute_has(parent, Query.t) :: parent
+
+  defmacro refute_has(parent, query) do
+    quote do
+      parent = unquote(parent)
+      query  = unquote(query)
+
+      case execute_query(parent, query) do
+        {:error, _not_found} ->
+          parent
+        {:ok, query} ->
```

Comment:
`execute_query` is used here to get the query with the result for a nice error message, above at `assert_has` I still use `has?` as I noticed that in the `error` clause of `execute_query` the second argument isn't the `query` so it doesn't exactly help there. At the same time, it breaks the symmetry between the two - not sure what to best do here. Input welcome :)